src/apps/competitions/tasks.py

```diff
@@ -744,3 +745,20 @@ def update_phase_statuses():
     competitions = Competition.objects.exclude(phases__in=Phase.objects.filter(is_final_phase=True, end__lt=now()))
     for comp in competitions:
         comp.update_phase_statuses()
+
+
+@app.task(queue='site-worker')
+def submission_status_cleanup():
+    submissions = Submission.objects.filter(status=Submission.RUNNING).exclude(has_children=True).select_related('phase')
+
+    for sub in submissions:
+        # Check if the submission has been running for 24 hours longer than execution_time_limit
+        if sub.started_when < now() - timedelta(milliseconds=(3600000 * 24) + sub.phase.execution_time_limit):
+            if sub.parent is not None:
+                app.control.revoke(sub.parent.celery_task_id, terminate=True)
+                sub.parent.status = Submission.FAILED
+                sub.parent.save()
+            else:
```

Comment:
don't we want to run this side of the else in both cases?

Reply:
..and the DRY violations here make me think we should have a method on `Submission` model that does this..

```python
class Submission...
    def fail_and_cancel_task(self):
            app.control.revoke(self.celery_task_id, terminate=True)
            self.status = Submission.FAILED
            self.save()
```

Reply:
I was assuming running this on the parent cancelled the child because I believe we use the `Submission.cancel()` method on only the parent submission when cancelling submissions run on multiple tasks.